### PR TITLE
GitHub/proposed/fixes for glibc 2.35

### PR DESCRIPTION
**Pull request by maximuska**

### CODE REVIEW

**Comment by maximuska** [comment]
done, thanks!

File: cbits/fs_override.c [hunk 1]
```diff
@@ -226,6 +228,22 @@ DEFINE_WRAPPER(DIR *, opendir, (const char *path))
...
+/* Depends on the full path */
+DEFINE_WRAPPER(DIR *, fdopendir, (int dirfd))
+{
+    initialize_process_state();
+    TRACE_DEBUG("handle fdopendir: %d", dirfd);
+    bool needs_await = false;
+    DEFINE_MSG(msg, opendir);
+    char fullpath[MAX_PATH];
+    if (!dereference_dir(dirfd, PS(fullpath))) {
+        return NULL;
```
Comment (maximuska) [reply cbits/fs_override.c:241]: done

Comment (sinelaw) [inline cbits/fs_override.c:241]: Consider setting errno = ENOENT

**Comment by sinelaw** [comment]
Tested on Centos 7, seems to work.

Notes:

To get buildsome to build, we had to:

1. Edit .stack/config.yaml to get stack to install the correct version of ghc-9.0.2 (see https://github.com/commercialhaskell/stack/issues/5561)

```
setup-info:
  ghc:
    linux64:
        9.0.2:
            url: "https://downloads.haskell.org/~ghc/9.0.2/ghc-9.0.2-x86_64-centos7-linux.tar.xz"
            content-length: 226498472
            sha1: 233a1b30112326d5d0e9a7bb646108582bc97197
            sha256: 42c78fa5ca8eb2bc7fabb8b9d7634d5b0ddd81c7fe16d8e212aeb9dac91a4f4e
```


2. Run the following:

```
sudo yum install libleveldb
sudo yum install centos-release-scl
sudo yum install devtoolset-7-gcc*
scl enable devtoolset-7 bash
stack install
```

**Pull request merged**